apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/configuration/AuthConfiguration.java

```diff
@@ -110,28 +110,56 @@ public LogoutHandler logoutHandler() {
-        AuthenticationManagerBuilder auth, DataSource datasource) throws Exception {
+    @Profile("mysql")
+    @Bean(name = "jdbcUserDetailsManager")
+    public static JdbcUserDetailsManager jdbcMysqlUserDetailsManager(PasswordEncoder passwordEncoder,
+                                                                AuthenticationManagerBuilder auth, DataSource datasource) throws Exception {
       JdbcUserDetailsManager jdbcUserDetailsManager = auth.jdbcAuthentication()
-          .passwordEncoder(passwordEncoder).dataSource(datasource)
-          .usersByUsernameQuery("select Username,Password,Enabled from `Users` where Username = ?")
-          .authoritiesByUsernameQuery(
-              "select Username,Authority from `Authorities` where Username = ?")
-          .getUserDetailsService();
+              .passwordEncoder(passwordEncoder).dataSource(datasource)
+              .usersByUsernameQuery("select Username,Password,Enabled from `Users` where Username = ?")
+              .authoritiesByUsernameQuery(
+                      "select Username,Authority from `Authorities` where Username = ?")
+              .getUserDetailsService();
 
       jdbcUserDetailsManager.setUserExistsSql("select Username from `Users` where Username = ?");
       jdbcUserDetailsManager
-          .setCreateUserSql("insert into `Users` (Username, Password, Enabled) values (?,?,?)");
+              .setCreateUserSql("insert into `Users` (Username, Password, Enabled) values (?,?,?)");
       jdbcUserDetailsManager
-          .setUpdateUserSql("update `Users` set Password = ?, Enabled = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
+              .setUpdateUserSql("update `Users` set Password = ?, Enabled = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
       jdbcUserDetailsManager.setDeleteUserSql("delete from `Users` where id = (select u.id from (select id from `Users` where Username = ?) as u)");
       jdbcUserDetailsManager
-          .setCreateAuthoritySql("insert into `Authorities` (Username, Authority) values (?,?)");
+              .setCreateAuthoritySql("insert into `Authorities` (Username, Authority) values (?,?)");
       jdbcUserDetailsManager
-          .setDeleteUserAuthoritiesSql("delete from `Authorities` where id in (select a.id from (select id from `Authorities` where Username = ?) as a)");
+              .setDeleteUserAuthoritiesSql("delete from `Authorities` where id in (select a.id from (select id from `Authorities` where Username = ?) as a)");
       jdbcUserDetailsManager
-          .setChangePasswordSql("update `Users` set Password = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
+              .setChangePasswordSql("update `Users` set Password = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
+
+      return jdbcUserDetailsManager;
+    }
+
+    @Profile("postgre")
```

Comment:
This duplicated the logic which doesn't look elegant as we may need to duplicate one for other database like h2, is there a better way to handle the sql difference?

Reply:
I didn't find a better way since we are direct using jdbc. Maybe we can refract this to use jpa. But I have no idea how to achieve this.

Reply:
It looks like the only difference is the quoted identifier, MySQL is \` while pg and h2 is ".
We may retrieve the `Dialect` and get the identifier from it?

```java
    @Bean
    public static JdbcUserDetailsManager jdbcUserDetailsManager(PasswordEncoder passwordEncoder,
        AuthenticationManagerBuilder auth, DataSource datasource,
        EntityManagerFactory entityManagerFactory) throws Exception {

      char openQuote = '`';
      char closeQuote = '`';
      try {
        SessionFactoryImplementor sessionFactory = entityManagerFactory.unwrap(
            SessionFactoryImplementor.class);
        Dialect dialect = sessionFactory.getJdbcServices().getDialect();
        openQuote = dialect.openQuote();
        closeQuote = dialect.closeQuote();
      } catch (Throwable ex) {
        //ignore
      }
      ...
    }


```

Reply:
@nobodyiam Fixed